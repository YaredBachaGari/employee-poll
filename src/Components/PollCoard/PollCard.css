.card-container{
    border: 0.5px solid #737272;
    margin: 1rem;
    text-align: center;
}
.card-text{
    border-bottom: 0.5px solid #737272;
}
.card-text p{
    padding: 0rem 5rem;
}
.card-container button{
    margin: 1rem 0;
    width: 90%;
    border: 0.5px solid #09c326;
    background: none;
    padding: 0.5rem;
    color: #09c326;
    border-radius: 5px;
    font-weight: bold;
}
.card-container button:hover{
    cursor: pointer;
    background-color: brown;
    color: aliceblue;
}
.username{
    font-weight: bold;
}
.polldate{
    color: #8c8c8c;
    font-size: 0.75rem;
}