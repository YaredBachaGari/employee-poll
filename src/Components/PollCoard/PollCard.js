import React from "react";
import "./PollCard.css";
import { formatDate, formatTime } from "../../utils/helper";
import { useNavigate } from "react-router-dom";

const PollCard = ({ survey, category }) => {
  const id = survey?.id;
  const author = survey?.author;
  const timestamp = survey?.timestamp;
  const SurveyDate = formatDate(timestamp);
  const SurveyTime = formatTime(timestamp);
  const navigate = useNavigate();
  const handleClick = () => {
    if (category === "New Questions") {
      if (id) {
        navigate(`/questions/${id}`);
      } else {
        navigate("/questions/notfound");
      }
    }
    if (category === "Done") {
      if (id) {
        navigate(`/result/${id}`);
      } else {
        navigate("/questions/notfound");
      }
    }
  };

  return (
    <div className="card-container">
      <div className="card-text">
        <p className="username" data-testid="author">{author}</p>
        <p className="polldate">
          <span data-testid="time">{SurveyTime}</span> | <span>{SurveyDate}</span>
        </p>
      </div>
      <button onClick={handleClick}>Show</button>
    </div>
  );
};

export default PollCard;
