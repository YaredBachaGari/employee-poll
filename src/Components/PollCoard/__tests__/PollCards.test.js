import React from "react";
import { render, screen } from "@testing-library/react";
import PollCard from "../PollCard";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import { useDispatch } from "react-redux";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

jest.mock("../../../Redux-handler/Actions/Users");
jest.mock("../../../Redux-handler/Actions/Questions");

describe("PollCard Component", () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should render username, date and show button", () => {
    const mockStore = configureStore();
    const store = mockStore({
      Questions: {
        data: {
          xyz: {
            id: "xyz",
            author: "john",
            timeStamp: 1467166872634,
            optionOne: {
              votes: ["alex", "sara"],
              text: "Build our new application with Javascript",
            },
            optionTow: {
              votes: ["brandon", "mattew"],
              text: "Build our new application with Asp.net",
            },
          },
        },
      },
    });
    const { card } = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollCard survey={store.Questions} category={"New Questions"} />
        </BrowserRouter>
      </Provider>
    );
    expect(card).toMatchSnapshot()
    expect(screen.getByText("Show")).toBeInTheDocument()
    expect(screen.getByTestId("author")).toBeInTheDocument()
    expect(screen.getByTestId("time")).toBeInTheDocument()
  });
});
