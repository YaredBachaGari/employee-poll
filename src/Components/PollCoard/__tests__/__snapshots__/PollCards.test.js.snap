// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollCard Component should render username, date and show button 1`] = `undefined`;
